File: typescript/src/server/address/format/restApiAddressFormatResponse.ts
import { RestApiResponse } from "../../restApiResponse";
import { RestApiResponseError } from "../../restApiResponseError";
import { RestApiAddressFormatEnrichment } from "./restApiAddressFormatEnrichment";
import { RestApiAddressFormatMetadata } from "./restApiAddressFormatMetadata";
import { RestApiFormatResult } from "./restApiFormatResult";

export type RestApiAddressFormatResponse = RestApiResponse & {
    error?: RestApiResponseError;
    result?: RestApiFormatResult;
    metadata?: RestApiAddressFormatMetadata;
    enrichment?: RestApiAddressFormatEnrichment;
};

File: csharp/DVSClient/DVSClient/server/address/format/RestApiAddressFormatResponse.cs
using System.Text.Json.Nodes;
using Newtonsoft.Json;

namespace DVSClient.Server.Address.Format
{
    [JsonObject(MemberSerialization.OptIn)]
    public class RestApiAddressFormatResponse : RestApiResponse
    {
        [JsonProperty("error")]
        public RestApiResponseError? Error { get; set; }

        [JsonProperty("result")]
        public RestApiFormatResult? Result { get; set; }

        [JsonProperty("metadata")]
        public RestApiAddressFormatMetadata? Metadata { get; set; }

        [JsonProperty("enrichment")]
        public RestApiAddressFormatEnrichment? Enrichment { get; set; }
    }
}